src/lib/blueprint.ts

```diff
@@ -47,10 +53,74 @@ interface Request {
   parameters: Parameter[]
 }
 
-interface Response {
+type Response = VoidResponse | ResourceResponse | ResourceListResponse
+
+interface BaseResponse {
   description: string
 }
 
+interface VoidResponse extends BaseResponse {
+  responseType: 'void'
+}
+
+interface ResourceResponse extends BaseResponse {
+  responseType: 'resource'
+  responseKey: string
+  resourceType: string
+}
+
+interface ResourceListResponse extends BaseResponse {
+  responseType: 'resource_list'
+  responseKey: string
+  resourceType: string
+}
+
+interface BaseProperty {
+  name: string
+  type: 'string' | 'enum' | 'record' | 'list' | 'object'
+  description?: string
+  isRequired?: boolean
+  isDeprecated?: boolean
+  deprecationMessage?: string
+}
+
+type Property =
+  | StringProperty
+  | EnumProperty
+  | RecordProperty
+  | ListProperty
+  | ObjectProperty
+
+interface StringProperty extends BaseProperty {
+  type: 'string'
+}
+
+interface EnumProperty extends BaseProperty {
+  type: 'enum'
+  values: EnumValue[]
+}
+
+interface EnumValue {
+  name: string
+  description: string
+  // Additional metadata fields for enum values
+}
+
+interface RecordProperty extends BaseProperty {
+  type: 'record'
+  properties: Property[]
+}
+
+interface ListProperty extends BaseProperty {
+  type: 'list'
+  items: Property
```

Comment:
Not sure what we should put here yet. Maybe `itemType: string` but let's leave it out and see how far we get with just description.
```suggestion
```